### PR TITLE
Build importer

### PR DESCRIPTION
**Pull request by sravfeyn**
Quick and dirty, but automated version of [adding builds to HQ](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/builds/README.md#adding-commcare-j2me-builds-to-commcare-hq). @millerdev 
Run `./manage.py commcare_build_importer` to start an interactive cli-session to import a build you need.

### CODE REVIEW

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+from cStringIO import StringIO
+from jenkinsapi.jenkins import Jenkins
+
+from corehq.apps.builds.models import CommCareBuild
+from django.core.management.base import BaseCommand, CommandError
+
+
+# supress requests library INFO logs
+logging.getLogger("requests").setLevel(logging.WARNING)
```
Comment (millerdev) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: This is a global setting, and will set it for the entire runtime (until changed elsewhere) if this module gets imported for any purpose. Maybe better to set it in [settings.py](https://github.com/dimagi/commcare-hq/blob/42f5cca2ade99b819a496f14c3af3201dd2b563d/settings.py#L847-L895) or possibly below in the `interactive_import` method?

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        print "Pinging Jenkins build server. Pelase wait..."
+        build_server = Jenkins("http://jenkins.dimagi.com")
+
+        # let user choose one of ['commcare-mobile-2.21', 'commcare-mobile-2.22']
+        jenkin_projects = build_server.keys()
+        selected_project_key = None
+        print "Jenkins has following projects. Please choose one"
+        print [p for p in jenkin_projects if 'commcare-mobile' in p]
+        while selected_project_key not in jenkin_projects:
+            selected_project_key = raw_input("")
```
Comment (millerdev) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: What if the user decides she wants to cancel? Might want to provide an option for that (empty string?).

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        # download and add the build
+        version_number = builds_by_version_number.get(selected_build_number)
+        build = selected_project.get_build_metadata(selected_build_number)
+        artifacts = build.get_artifact_dict()
+        artifacts_url = artifacts['artifacts.zip'].url
+
+        print "Downloading and importing artifacts.zip. Please wait..."
+        zip_file = requests.get(artifacts_url)
+        self.add_build(StringIO(zip_file.content), version_number, selected_build_number)
```
Comment (millerdev) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: The first argument you're passing in here is a file-like object, not a path. Maybe rename `build_path` to `build_file` below?

**Comment by sravfeyn** [comment]
@esoergel

File: corehq/apps/builds/README.md [hunk 4]
```diff
@@ -37,6 +37,10 @@ download and install your own locally built CommCare application!
 Adding CommCare (J2ME) Builds to CommCare HQ
 =====================================
 
+Following is a manual process to find and import a build. Alternatively, you can run
+`./manage.py commcare_build_importer` to do the same without leaving your console.
+This will run you through all the builds and let you import the build you need.
```
Comment (esoergel) [inline corehq/apps/builds/README.md]: :balloon: :cake: :fireworks:

File: requirements/dev-requirements.txt [hunk 5]
```diff
@@ -8,6 +8,7 @@ git+git://github.com/dimagi/couchdbkit-debugpanel.git
 Fabric==1.10.0
 ansible==1.8.2  # to run fab development *
 Sphinx==1.2b1
+jenkinsapi=0.2.28
```
Comment (esoergel) [inline requirements/dev-requirements.txt]: This needs to be a `==`

Comment (esoergel) [inline requirements/dev-requirements.txt]: I'm trying this out locally, so I just pushed a fix for that and the merge conflict.

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            print "Failed to fetch artifacts.zip at URL`"
+            print artifacts_url
+            return
+
+        self.add_build(StringIO(zip_file.content), version_number, selected_build_number)
+
+    @property
+    @memoized
+    def jenkin_projects(self):
+        print "Pinging Jenkins build server. Pelase wait..."
```
Comment (esoergel) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: Minor typo here: `Pelase`

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        logging.getLogger("requests").setLevel(logging.WARNING)
+
+        # let user chose a project from the list ['commcare-mobile-2.21', ...]
+        selected_project_key = None
+        print "Jenkins has following projects. Please choose one (to end enter END)"
+        print [p for p in self.jenkin_projects if 'commcare-mobile' in p]
+        while selected_project_key not in self.jenkin_projects:
+            selected_project_key = raw_input("")
+            if selected_project_key.lower() == 'end':
+                return
```
Comment (esoergel) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: It'd be nice to give some feedback if the `selected_project_key` is not found - I typed `"commcare-mobile"` (with the quotes) and it just hung there, which was confusing

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    def build_server(self):
+        return Jenkins("http://jenkins.dimagi.com")
+
+    def add_build(self, builf_file, version, build_number):
+        try:
+            CommCareBuild.create_from_zip(builf_file, version, build_number)
+        except Exception as e:
+            raise CommandError("%s" % e)
+        self.stdout.write('Build %s #%s created\n' % (version, build_number))
+        self.stdout.write('You can see a list of builds at [your-server]/builds/')
```
Comment (esoergel) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: You can get the whole URL using `corehq.utils.view_utils.absolute_reverse` (although it looks like you'd need to name the corresponding URLs).  Alternatively, `get_url_base()` will do the trick.

File: corehq/apps/builds/management/commands/commcare_build_importer.py [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    def handle(self, *args, **options):
+        if not args:
+            self.interactive_import()
+        else:
+            raise CommandError(self.help)
+
+    def interactive_import(self):
+
+        # supress requests library INFO logs
+        logging.getLogger("requests").setLevel(logging.WARNING)
```
Comment (esoergel) [inline corehq/apps/builds/management/commands/commcare_build_importer.py]: These are so annoying, I might need to do that everywhere.

**Comment by esoergel** [comment]
I had a couple small comments, but it looks good overall.  I tried it out locally and everything worked.  I really like the interactive approach over looking it up yourself and specifying a bunch of flags - very intuitive and easy to use.

One thing that might make it a little friendlier would be to number the options, so you see:
```
Jenkins has following projects. Please choose one (to end enter END)
Pinging Jenkins build server. Please wait...
1: commcare-mobile
2: commcare-mobile-2.13
3: commcare-mobile-2.19
4: commcare-mobile-2.20
5: commcare-mobile-2.21
6: commcare-mobile-2.22
7: commcare-mobile-2.23
8: commcare-mobile-non-blocking-update
```
and type `1` instead of typing the whole thing out - same for selecting the build number.

**Comment by sravfeyn** [comment]
These are great points! Thanks for noting all them down. 

I didn't want to put more time on this (I sneaked this in an interrupt week in PID way), so I made just the mvp. More usability tweaks would be great, but I feel it's better to just push it out (as long as this feels usable) and let everyone add improvements incrementally whenever an annoyance is found, specially since this feels like an improvement over existing process anyway.

I will see which ones I can get to.

**Comment by esoergel** [comment]
Fair point.  I'll merge as-is, then you can sneak any changes you feel like making in later in a smaller PR.

**Comment by esoergel** [comment]
Also, nice PID win!

**Comment by benrudolph** [comment]
cool! look forward to using this!

**Pull request merged**